docs/api_comparison.md

```diff
@@ -0,0 +1,254 @@
+# API comparison
+
+The `h3-py` library provides several APIs which share the same functions,
+but differ in their input and output types:
+
+- `h3.api.basic_str`
+- `h3.api.basic_int`
+- `h3.api.numpy_int`
+- `h3.api.memview_int`
+
+## Why are there multiple APIs?
+
+<!-- TODO: if i make this a notebook, do i get to have the headings i want?
+BETTER: keep as markdown, but let the launcher convert the markdown to a notebook for people to run and test! -->
+
+Under the hood in the C library, H3 indices are represented as
+unsigned 64-bit integers (`uint64_t`), and collections of indexes
+are represented with C arrays.
+
+For human readability and interpretability, we'll often represent integer
```

Comment:
Updated the wording a bit, but I'm trying to avoid the absolute because we'll sometimes "represent" indices as integers (like with `basic_int`) and sometimes as strings (like with `basic_str`).

But maybe it's a question of whether "represent" means "in the docs" or "in the library"?